### ISSUE TITLE
Error : "Expecting single root statement in program"?

### ISSUE THREAD

**Issue by zabojad** [jasononeil/webpack-haxe-loader#7]
Hi!

I'm trying the haxe-loader for the first time. I am facing the below issue when trying to run my project in webpack-dev-server. Any idea why?
```
$ node_modules/webpack/bin/webpack.js --config webpack.config.dev.js
haxe -D webpack_namespace=build-web -main myproj.Boot -js /var/folders/mc/2kyt0cbn12d95zcg5wzb0sj80000gn/T/tmp-56505G9un2hNdyd9X.js -cp src -lib js-kit -lib haxe-loader -lib react -lib react-addons -debug

/Users/thomas/wksp_myproj/myproj/node_modules/haxe-modular/tool/bin/split.js:400
				throw new js__$Boot_HaxeError("Expecting single root statement in program");
				^
Error: Expecting single root statement in program
    at Object.walkProgram (/Users/thomas/wksp_myproj/myproj/node_modules/haxe-modular/tool/bin/split.js:400:11)
    at Object.processInput (/Users/thomas/wksp_myproj/myproj/node_modules/haxe-modular/tool/bin/split.js:335:8)
    at Object.Parser (/Users/thomas/wksp_myproj/myproj/node_modules/haxe-modular/tool/bin/split.js:325:7)
    at Object.Main.run.$hx_exports.run [as run] (/Users/thomas/wksp_myproj/myproj/node_modules/haxe-modular/tool/bin/split.js:311:15)
    at processOutput (/Users/thomas/wksp_myproj/myproj/node_modules/haxe-loader/index.js:68:27)
    at exec (/Users/thomas/wksp_myproj/myproj/node_modules/haxe-loader/index.js:46:27)
    at ChildProcess.exithandler (child_process.js:197:7)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:877:16)
    at Socket.<anonymous> (internal/child_process.js:334:11)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at Pipe._handle.close [as _onclose] (net.js:498:12)
```

**Comment by zabojad** [comment]
If I comment the lines 400 -> 402 in `node_modules/haxe-modular/tool/bin/split.js`, it works. But I'm not sure what those lines were supposed to do...

```
else {
    throw new js__$Boot_HaxeError("Expecting single root statement in program");
}
```

**Comment by jasononeil** [comment]
@elsassph do you want to comment on this?  (@zabojad, so you know split.js comes from the haxe-modular repo and Philippe knows much more about that then I do!)

Could you also post the hxml file you're using?

**Comment by zabojad** [comment]
Hi there !

My hxml is pretty much standard:
```
-main myproj.Boot
-js app/myproj.js
-cp src
-lib js-kit
-lib haxe-loader
-lib react
-lib react-addons
-debug
```

**Comment by elsassph** [comment]
What version Haxe are you using?

**Comment by zabojad** [comment]
```
$ haxe -v
Haxe Compiler 3.4.2 - (C)2005-2017 Haxe Foundation
```
I can't install the `3.4.3` yet [because of](https://groups.google.com/forum/#!topic/haxelang/M3i9a7u6X_s)

**Comment by elsassph** [comment]
Can you share a sample project?

**Comment by elsassph** [comment]
This is fixed in haxe-modular now, requires an update of the modular dependency to 0.6.0 in haxe-loader.

**Comment by zabojad** [comment]
Nice, thank you Philippe! Sorry for not having answered earlier...

**Comment by elsassph** [comment]
I hope that's the solution though...

**Comment by zabojad** [comment]
I'll come back to it at the end of this week and confirm weither it's fixed or not after having updated the haxe-modular. By the way, how should I do this? If haxe-modular is already installed for my project, then haxe-loader will use it or will it install its own version of haxe-modular?

**Comment by elsassph** [comment]
You can point on my fork of haxe-loader to try it

**Comment by jasononeil** [comment]
@zabojad this is now released as `haxe-loader@0.4.1`

**Comment by zabojad** [comment]
@jasononeil @elsassph just for your information, the issue doesn't show up anymore with `haxe-loader@0.4.1`
